src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -4424,19 +4442,101 @@ def generate_agg_qc(
                 )
 
             else:
-                for agg_func, cols in agg_func_to_col_map.items():
-                    col_single_agg_func_map = {
-                        column: AggFuncInfo(
-                            func=agg_func if column in cols else "min",
-                            is_dummy_agg=column not in cols,
+                if uses_named_aggs:
+                    # If this is true, then we are dealing with agg with NamedAggregations.
+
+                    def generate_single_agg_column_func_map(
+                        identifier_pair: PandasLabelToSnowflakeIdentifierPair,
+                        agg_func: AggFuncWithLabel,
+                    ) -> dict[PandasLabelToSnowflakeIdentifierPair, AggFuncInfo]:
+                        """
+                        Helper function to produce the agg func map for a single aggregation, including dummy aggregations.
```

Comment:
Can you explain why you need this function here? it seems that with named agg, duplicated name is not allowed, and each named agg can only have one function. You should be able to concatenate them without the need the dummy agg? do you have an example/test explains why you need to do it like this?

Reply:
the dummy aggs are necessary for when different aggs are on different columns - e.g. say we have columns A, B, C, and aggregations x=('A', 'min'), y=('C', 'max'). When we do the aggregations for 'A', we need to pass in the dictionary {('A', '"A"'): AggFuncInfo('min', 'x'), ('C', '"C"'): AggFuncInfo('min', 'y', is_dummy_agg=True)} - this function is just a helper function to loop through all columns being aggregated on, and add dummy_aggs for the columns that the current aggregation is on, the same as the dict comprehension in line 4530 - I could make this a dict comprehension too, or change line 4530 to use this function - do you have a preference for which to do?

Reply:
resolved offline.